controllers/sriovnetworkpoolconfig_controller.go

```diff
@@ -175,14 +176,24 @@ func (r *SriovNetworkPoolConfigReconciler) syncOvsHardwareOffloadMachineConfigs(
 				return fmt.Errorf("couldn't delete MachineConfig: %v", err)
 			}
 		} else {
-			if bytes.Equal(foundMC.Spec.Config.Raw, mc.Spec.Config.Raw) {
+			var foundIgn, renderedIgn interface{}
+			// The Raw config JSON string may have the fields reordered.
+			// For example the "path" field may come before the "contents"
+			// field in the rendered ignition JSON; while the found
+			// MachineConfig's ignition JSON would have it the other way around.
+			// Thus we need to unmarshal the JSON for both found and rendered
+			// ignition and compare.
+			json.Unmarshal(foundMC.Spec.Config.Raw, &foundIgn)
+			json.Unmarshal(mc.Spec.Config.Raw, &renderedIgn)
```

Comment:
Can't we deep compare the Spec? iow, something like

```
if !reflect.DeepEqual(foundMC.spec, foundMC.spec,

Reply:
I believe DeepEqual will always return false if we directly use the spec (I gave this a try). The reason is due to the Raw JSON config having fields reordered. This will cause the comparison to return not equal even though the fields are the same.

Reply:
Then how come https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/master/controllers/sriovibnetwork_controller.go#L155 doesn't have that issue? Did you investigate that?

Reply:
I tried reflect.DeepEqual(mc.Spec.Config, foundMC.Spec.Config) and reflect.DeepEqual(mc.Spec, foundMC.Spec); the result for both is always false.

Reply:
I think it is the type "runtime.RawExtension" that is causing this.

Reply:
I think you are right here @wizhaoredhat that type is just a []byte